packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -519,6 +519,14 @@ - (FlutterError *_Nullable)onListenWithArguments:(id _Nullable)arguments
 /// is useful for the case where the Engine is in the process of deconstruction
 /// so the channel is going to die or is already dead.
 - (void)disposeSansEventChannel {
+  // This check prevents the crash caused by removing the KVO observers twice.
+  // When performing a Hot Restart, the leftover players are disposed directly
+  // and also receive onTextureUnregistered: callback leading to possible
```

Comment:
nit: just wanna be clearer

"the leftover players are disposed twice - once directly (from which function?), and once when receiving onTextureUnregistered callback"

Reply:
@hellohuanlin I've updated the comment yet again to make it clear